src/database/DataUpdater.py

```diff
@@ -7,8 +7,8 @@
 # Not abstract, so a user can choose to default what database to interact with using this Executor
 class DataUpdater(iDatabaseExecutor):
 
-    def __init__(self, *parents):
-        super().__init__(*parents)
+    def __init__(self, parents=[]):
```

Comment:
Careful with this and all the other default list arguments. See https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument for details on how this could be a problem if we ever modify a default parent list. Doesn't look like we ever do that in our codebase so it should be fine, but changing the default to None and doing a None check in iExecutor would be safer in the long run.

Reply:
oh interesting, I'll make that change